topics/howtosim.rst

```diff
@@ -0,0 +1,255 @@
+   disk_mass            : XXXXX                # Mass of stellar disk (e.g. 0.0125)
+   dtype                : exponential          # Disk type for condition (one of: constant, gaussian, mn, exponential)
+   expcond              : true                 # Use analytic density function for computing EmpCylSL basis
+   gentype              : asymmetric           # DiskGenType string for velocity initialization (Jeans, Asymmetric, or Epicyclic)
+   halofile1            : XXXXX                # File with input halo model (your .model file here)
+   evolved              : false
+   hbods                : halo.bods            # Halo particle output file
+   ignore               : false                # Recompute EOF regardless of existence
+   multi                : false                # Use multimass halo
+   ndisk                : XXXXX                # Number of disk particles (e.g. 1000000)
+   ngas                 : 0                    # Number of gas particles
+   ngparam              : 3                    # Number of gas particle parameters
+   nhalo                : XXXXX                # Number of halo particles (e.g. 10000000)
+   report               : true                 # Report particle progress in EOF computation
+   runtag               : XXXXX                # Label prefix for diagnostic images (e.g. 'run1')
+   scale_height         : XXXXX                # Scale height for disk realization (e.g. 0.002)
+   scale_length         : XXXXX                # Scale length for disk realization (e.g. 0.01)
+   threads              : 1                    # Number of lightweight threads
+   zerovel              : true                 # zero center of mass and velocity
+   suffix               : ""
+
+Parameters that have values set to 'XXXXX' are those that users are encouraged to change. Example values are given in the
+comment following the parameter value. The remaining parameters are "expert parameters" and generally do not need to be changed
+unless you have a good reason to use a different value. Note that the ``exponential`` disk is
+exponential in R and :math:`sech^2` in z.
+
+To generate the initial conditions, you would run:
+
+.. code-block:: python
+
+   mpirun gendisk --config gendisk_step1.yml 
+
+in the terminal, replacing ``gendisk_step1.yml`` with the name of your YAML file that specifies the ``gendisk`` parameters. If
+you set ``ignore = false``, it may take a moment to generate the cylindrical bases cache file. The cylindrical bases will be
+saved out to whatever name is specified in ``cachefile``, and the halo basis will generally be saved out to
+``.slgrid_sph_cache``. Note that the ``diskgen`` YAML file here is distinct from the EXP YAML config file, EXP YAML config files 
+look like :ref:`this annotated example <yamlconfig>`.
+
+Step Two: generate cache files
+******************************
+If you didn't generate cache files in step one, you need to generate these files now. An example of how to do this is in the
```

Comment:
Similar to my comments above, the user doesn't have to generate bases (and associated cache files) prior to running `exp`; `exp` will do it if it doesn't find a cache file.

Reply:
Ah - that was not clear to me, I thought that you had to generate them before running. This is good to know!

Reply:
It is also true that some bases don't have cache files.  Like `cube`.   Also `bessel`, `c_brock`, `c_brock_disk`, and `hernquist` but those are slated for removal at this point.   The new `slab` basis has a cache, but like `sphere`, I've been letting `exp` remake those on the fly as necessary, since they are quick.  The only one that takes significant time is `cylinder`.  The `flatdisk` basis (e.g. razor thin, exponential or Zang) also takes some time.

Reply:
Actually - it looks like `exp` won't generate an EOF cache file on its own, it needs to be fed one. The SL cache file is generated on-the-fly if given a name, but once `exp` gets to the part where it checks the EOF cachefile, it quits with:

`Cylinder: can not read explicitly specified EOF file <NAMEHERE>
Cylinder: shamelessly aborting . . .
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.`

Same thing occurs if nothing is passed to `cachename` in the disk config parameters

Reply:
I think you've got an older/different version.  I changed the policy from 'shameless' to backup in the recent devel branch which was merged a week or so ago.